### PR TITLE
Car Port: Prius 2021 TSS2

### PR DESCRIPTION
**Pull request by rav4kumar**
**Checklist**
- [done ] added to README
- [unban me so I can get route real quick and ban me back] test route added to [test_car_models](../../selfdrive/test/test_car_models.py)
- [done] route with openpilot:
- [ done] route with stock system:

### CODE REVIEW

**Comment by rav4kumar** [comment]
this is hand tune from corolla from @Spairrow326

**Comment by pd0wm** [comment]
How does the regular TSS2 PID tuning perform?

**Comment by ShaneSmiskol** [comment]
Steer actuator delay should probably be removed since it's TSS2 and might not have that large of a delay (similar to the other Toyotas with 0.12)

**Comment by rav4kumar** [comment]
> How does the regular TSS2 PID tuning perform?

its really delayed. almost feels like stock LTA bounces back it forward.

File: selfdrive/test/test_car_models.py [hunk 1]
```diff
@@ -484,6 +484,7 @@ def get_route_log(route_name):
   TOYOTA.HIGHLANDER,
   TOYOTA.HIGHLANDERH,
   TOYOTA.HIGHLANDERH_TSS2,
+  TOYOTA.PRIUS_TSS2,
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:487]: we'll need a route before merging this

Comment (rav4kumar) [reply selfdrive/test/test_car_models.py:487]: Can you unban me so I can get route and then ban me back if you would like. I am just trying to upstream the fp.

Comment (VirtuallyChris) [reply selfdrive/test/test_car_models.py:487]: What's your dongle ID?

**Comment by rav4kumar** [comment]
0a0de17a1e6a2d15 @VirtuallyChris 

https://streamable.com/3l9io1 this is what happens right now. maybe you guys blocked my account or imei because changing account and registration does the same

**Comment by VirtuallyChris** [comment]
Your device isn't banned, it looks like there's just a pairing bug in the app. Can you let me know what email you are trying to pair, and I will do it?

**Comment by rav4kumar** [comment]
@VirtuallyChris I sent you email! Thank you!

**Comment by rav4kumar** [comment]
@adeebshihadeh @pd0wm https://my.comma.ai/cabana/?route=0a0de17a1e6a2d15%7C2020-09-21--21-24-41&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F0a0de17a1e6a2d15%2F53d66731e566a19fd9db9757135ea460_2020-09-21--21-24-41&seekTime=39 there you go. can you guys also help tune it and get new dbc! that would be amazing

**Comment by adeebshihadeh** [comment]
New car ports are done by the community, but I'd recommend starting with PID and tuning that. It doesn't need to be perfect.

File: selfdrive/car/toyota/values.py [hunk 2]
```diff
@@ -265,6 +267,9 @@ class CAR:
...
+  }],
+  CAR.PRIUS_TSS2: [{
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:271]: We also need the FW fingerprints added and the car should be added to the IGNORED_FINGERPRINTS list

File: selfdrive/car/toyota/values.py [hunk 3]
```diff
@@ -39,6 +40,7 @@ class CAR:
   LEXUS_CTH = "LEXUS CT 200H 2018"
   RAV4H_TSS2 = "TOYOTA RAV4 HYBRID 2019"
   LEXUS_NXH = "LEXUS NX300H 2018"
+
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:43]: revert this

**Comment by VirtuallyChris** [comment]
> New car ports are done by the community, but I'd recommend starting with PID and tuning that. It doesn't need to be perfect.

I'd agree with this TSS2 does really well with PID tuning. Maybe take the 2020 Corolla tune and work from there?

**Comment by rav4kumar** [comment]
@adeebshihadeh i dont have black panda to get fw.

**Comment by rav4kumar** [comment]
https://media.discordapp.net/attachments/533837932457361419/758044414126850098/unknown.png

**Comment by rav4kumar** [comment]
https://cdn.discordapp.com/attachments/560976017536319491/760003596464160799/Screenshot_114.png safety at 55 looks right. should i create new dbc with `0.55` or change the current one? i am sure all the tss2 is about the same. 73 looks way too off.

**Comment by pd0wm** [comment]
I doubt all the TSS2 cars are wrong. Looks like they did something special for the prius again, in that case we need a separate DBC file.

**Comment by rav4kumar** [comment]
@pd0wm here is 2020 Corolla LE this is at 73 https://prnt.sc/upgv3i and here at 55 https://prnt.sc/upgvfy

**Comment by rav4kumar** [comment]
@pd0wm @adeebshihadeh can you guys let me what else i need to do add to get this pr merged please and thank you!

**Comment by pd0wm** [comment]
We need FW versions. We can't use the new can fingerprint for fingerprinting purposes since it might break existing cars.

In the meantime I'll write a script to more quickly verify the safety params. Then I'll go over existing TSS2 cars as well to see what the correct value is.

**Comment by rav4kumar** [comment]
@pd0wm fw added! were you able to verify the safetyprams?

**Comment by adeebshihadeh** [comment]
Looks like the TSS2 DBC may be wrong, but for this PR, let's use the current factor. I'll write a unit test for this for all the toyotas and update factors as necessary.

Remaining TODOs:
* add new prius to IGNORED_FINGERPRINTS
* use current safety param and DBC

File: README.md [hunk 4]
```diff
@@ -107,7 +107,9 @@ Supported Cars
 | Toyota    | Highlander 2020               | All               | openpilot        | 0mph               | 0mph              |
 | Toyota    | Highlander Hybrid 2017-19     | All               | Stock<sup>3</sup>| 0mph               | 0mph              |
 | Toyota    | Highlander Hybrid 2020        | All               | openpilot        | 0mph               | 0mph              |
-| Toyota    | Prius 2016-20                 | TSS-P             | Stock<sup>3</sup>| 0mph               | 0mph              |
+| Toyota    | Prius 2016                    | TSS-P             | Stock<sup>3</sup>| 0mph               | 0mph              |
+| Toyota    | Prius 2017-20                 | All               | Stock<sup>3</sup>| 0mph               | 0mph              |
```
Comment (VirtuallyChris) [inline README.md:111]: Revert these readme changes and only keep the Prius 2021 addition

**Comment by rav4kumar** [comment]
done @adeebshihadeh can we get it merged on 0710 release that would be amazing!

**Comment by adeebshihadeh** [comment]
It's too late to get into 0.7.10, but it will be in the next release.

**Pull request merged**